`vp i <package>` Results in `EPERM: operation not permitted` With PNPM Workspace `catalog: prefer`

### Describe the bug

This occurs when using `pnpm` as the package manager and [`catalogMode: prefer`](https://pnpm.io/catalogs#catalogmode) in `pnpm-workspace.yaml`.

When running a development server (like a Vite web app), Vite Plus locks `pnpm-workspace.yaml`, making new package installations in `/apps/*/package.json` or in the workspace root fail with this error:

> `EPERM: operation not permitted, rename 'C:\Users\admin\Desktop\code\repros\repro-vp-dev-locking-pnpm-workspace\pnpm-workspace.yaml.1955691642' -> 'C:\Users\admin\Desktop\code\repros\repro-vp-dev-locking-pnpm-workspace\pnpm-workspace.yaml'`

But the `package.json` and `pnpm-lock.yaml` changes.

Subsequent installs (without reverting the `package.json` and `pnpm-lock.yaml` changes) results in this error:

> `ERR_PNPM_CATALOG_ENTRY_NOT_FOUND_FOR_SPEC No catalog entry 'is-even' was found for catalog 'default' `

Because the `package.json` updates with the package with a `catalog:` version BUT without a corresponding version entry in `pnpm-workspace.yaml`.

### Other Build Tool Behaviors

In other frameworks like Nuxt, Next.js, Vite React (without Vite+), you can install dependencies during local development just fine, so it's a point of friction in terms of DX when not having the same capabilities in Vite+.

### Reproduction

https://github.com/Luzefiru/repro-vp-dev-locking-pnpm-workspace

### Steps to reproduce

Reproduced using the default `vp create` monorepo template.

## Create the Repo & Start A Dev Server

```bash
admin@windows MINGW64 ~/Desktop/code/repros
$ vp create
VITE+ - The Unified Toolchain for the Web

  Vite+ Monorepo

◇ Package name:
  repro-vp-dev-locking-pnpm-workspace

◇ Which package manager would you like to use?
  pnpm

◇ Which agents are you using?
  GitHub Copilot

◇ Which editor are you using?
  VSCode

◇ Set up pre-commit hooks to run formatting, linting, and type checking with auto-fixes?
  Yes

◇ Scaffolded repro-vp-dev-locking-pnpm-workspace with Vite+ monorepo
• Node 24.14.1  pnpm 10.33.0
✓ Dependencies installed in 8.7s
→ Next: cd repro-vp-dev-locking-pnpm-workspace && vp run

admin@windows MINGW64 ~/Desktop/code/repros/repro-vp-dev-locking-pnpm-workspace (main)
$ vp i
VITE+ - The Unified Toolchain for the Web

Scope: all 3 workspace projects
Lockfile is up to date, resolution step is skipped
Already up to date
. prepare$ vp config
└─ Done in 250ms
Done in 1.1s using pnpm v10.33.0

admin@windows MINGW64 ~/Desktop/code/repros/repro-vp-dev-locking-pnpm-workspace (main)
$ vp run dev
VITE+ - The Unified Toolchain for the Web

~/apps/website$ vp dev ⊘ cache disabled

  VITE+ v0.1.16

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
```

## Try Installing Packages in Another Terminal
Then, in another terminal, I attempt to install a package:

```bash
admin@windows MINGW64 ~/Desktop/code/repros/repro-vp-dev-locking-pnpm-workspace (main)
$ cd apps/website/

admin@windows MINGW64 ~/Desktop/code/repros/repro-vp-dev-locking-pnpm-workspace/apps/website (main)
$ vp i is-even
VITE+ - The Unified Toolchain for the Web

../..                                    | Progress: resolved 152, reused 0, downloaded 0, added 0, done

dependencies:
+ is-even catalog:

../..                                    |   +5   -2 +-
../.. prepare$ vp config
└─ Done in 255ms
 EPERM  EPERM: operation not permitted, rename 'C:\Users\admin\Desktop\code\repros\repro-vp-dev-locking-pnpm-workspace\pnpm-workspace.yaml.4100967199' -> 'C:\Users\admin\Desktop\code\repros\repro-vp-dev-locking-pnpm-workspace\pnpm-workspace.yaml'
 # (!!!) this is the error
 $ git status
On branch main
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   package.json
        modified:   ../../pnpm-lock.yaml

no changes added to commit (use "git add" and/or "git commit -a")
 # (!!!) changes are made to package.json
```

The `package.json` correctly shows `dependencies` with `"is-even": "catalog:"`, but the `pnpm-workspace.yaml` does not include the entry for it with the version specifier.

Afterwards, any attempt to install it again will lead to this error:

```bash
admin@windows MINGW64 ~/Desktop/code/repros/repro-vp-dev-locking-pnpm-workspace/apps/website (main)
$ vp i is-even
VITE+ - The Unified Toolchain for the Web

 ERR_PNPM_CATALOG_ENTRY_NOT_FOUND_FOR_SPEC  No catalog entry 'is-even' was found for catalog 'default'.
../..                                    | Progress: resolved 6, reused 0, downloaded 0, added 0
```

This is also true for workspace root-level installations:

```bash
admin@windows MINGW64 ~/Desktop/code/repros/repro-vp-dev-locking-pnpm-workspace (main)
$ git reset --hard
HEAD is now at a25119c feat: init Monorepo with vp create pnpm, GitHub Copilot, VSCode, Yes

admin@windows MINGW64 ~/Desktop/code/repros/repro-vp-dev-locking-pnpm-workspace (main)
$ vp i -w is-even
VITE+ - The Unified Toolchain for the Web

Progress: resolved 152, reused 0, downloaded 0, added 0, done

dependencies:
+ is-even catalog:

Already up to date
. prepare$ vp config
└─ Done in 227ms
 EPERM  EPERM: operation not permitted, rename 'C:\Users\admin\Desktop\code\repros\repro-vp-dev-locking-pnpm-workspace\pnpm-workspace.yaml.1955691642' -> 'C:\Users\admin\Desktop\code\repros\repro-vp-dev-locking-pnpm-workspace\pnpm-workspace.yaml'
```

## Workaround

Stopping the `vp run dev` local development server unlocks the file from edits and allows you to run `vp i <package>` successfully.

### System Info

```shell
admin@windows MINGW64 ~/Desktop/code/repros/repro-vp-dev-locking-pnpm-workspace (main)
$ vp env current
VITE+ - The Unified Toolchain for the Web

Environment:
  Version       24.14.1
  Source        engines.node
  Source Path   C:\Users\admin\Desktop\code\repros\repro-vp-dev-locking-pnpm-workspace\package.json
  Project Root  C:\Users\admin\Desktop\code\repros\repro-vp-dev-locking-pnpm-workspace

Tool Paths:
  node  C:\Users\admin\.vite-plus\js_runtime\node\24.14.1\node.exe
  npm   C:\Users\admin\.vite-plus\js_runtime\node\24.14.1\npm.cmd
  npx   C:\Users\admin\.vite-plus\js_runtime\node\24.14.1\npx.cmd

admin@windows MINGW64 ~/Desktop/code/repros/repro-vp-dev-locking-pnpm-workspace (main)
$ vp --version
VITE+ - The Unified Toolchain for the Web

vp v0.1.16

Local vite-plus:
  vite-plus  v0.1.16

Tools:
  vite             v8.0.5
  rolldown         v1.0.0-rc.13
  vitest           v4.1.2
  oxfmt            v0.43.0
  oxlint           v1.58.0
  oxlint-tsgolint  v0.20.0
  tsdown           v0.21.7

Environment:
  Package manager  pnpm v10.33.0
  Node.js          v24.14.1 (engines.node)
```

### Used Package Manager

pnpm


### Validations

- [x] Read the [Contributing Guidelines](https://github.com/voidzero-dev/vite-plus/blob/main/CONTRIBUTING.md).
- [x] Check that there isn't [already an issue](https://github.com/voidzero-dev/vite-plus/issues) for the same bug.
- [x] Confirm this is a Vite+ issue and not an upstream issue (Vite, Vitest, tsdown, Rolldown, or Oxc).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example).

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -89,7 +89,7 @@ dunce = "1.0.5"
 fast-glob = "1.0.0"
 flate2 = { version = "=1.1.9", features = ["zlib-rs"] }
 form_urlencoded = "1.2.1"
-fspy = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "076cef486127e6cd1fefc58945f00dac316888ca" }
+fspy = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "954cd3bead71be4ddc1e3183110d713bbe26286b" }
 futures = "0.3.31"
 futures-util = "0.3.31"
 glob = "0.3.2"
@@ -194,16 +194,16 @@ vfs = "0.13.0"
 vite_command = { path = "crates/vite_command" }
 vite_error = { path = "crates/vite_error" }
 vite_js_runtime = { path = "crates/vite_js_runtime" }
-vite_glob = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "076cef486127e6cd1fefc58945f00dac316888ca" }
+vite_glob = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "954cd3bead71be4ddc1e3183110d713bbe26286b" }
 vite_install = { path = "crates/vite_install" }
 vite_migration = { path = "crates/vite_migration" }
 vite_setup = { path = "crates/vite_setup" }
 vite_shared = { path = "crates/vite_shared" }
 vite_static_config = { path = "crates/vite_static_config" }
-vite_path = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "076cef486127e6cd1fefc58945f00dac316888ca" }
-vite_str = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "076cef486127e6cd1fefc58945f00dac316888ca" }
-vite_task = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "076cef486127e6cd1fefc58945f00dac316888ca" }
-vite_workspace = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "076cef486127e6cd1fefc58945f00dac316888ca" }
+vite_path = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "954cd3bead71be4ddc1e3183110d713bbe26286b" }
+vite_str = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "954cd3bead71be4ddc1e3183110d713bbe26286b" }
+vite_task = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "954cd3bead71be4ddc1e3183110d713bbe26286b" }
+vite_workspace = { git = "https://github.com/voidzero-dev/vite-task.git", rev = "954cd3bead71be4ddc1e3183110d713bbe26286b" }
 walkdir = "2.5.0"
 wax = "0.6.0"
 which = "8.0.0"

